crates/hir/src/hir_def/module_tree.rs

```diff
@@ -110,6 +110,7 @@ impl<'db> ModuleTree<'db> {
 /// top level modules. This function only depends on an ingot structure and
 /// external ingot dependency, and not depends on file contents.
 #[salsa::tracked(return_ref)]
+#[allow(elided_named_lifetimes)]
```

Comment:
Clippy and the rust compiler were battling back and forth, on the one hand Clippy would complain about `needless_lifetimes` and change it to the elided version, but then the rust compiler would complain about `elided_named_lifetimes`.  So I added this allow rule here and one other place.